examples/features/customloadbalancer/client/customroundrobin/customroundrobin.go

```diff
@@ -0,0 +1,178 @@
+}
+
+func (customRoundRobinBuilder) Name() string {
+	return customRRName
+}
+
+func (customRoundRobinBuilder) Build(cc balancer.ClientConn, bOpts balancer.BuildOptions) balancer.Balancer {
+	crr := &customRoundRobin{
+		ClientConn: cc,
+		bOpts:      bOpts,
+	}
+	crr.balancerAggregator = balanceraggregator.Build(crr, bOpts)
+	return crr
+}
+
+var logger = grpclog.Component("example")
+
+type customRoundRobin struct {
+	// All state and operations on this balancer are either initialized at build
+	// time and read only after, or are only accessed as part of its
+	// balancer.Balancer API (UpdateState from children only comes in from
+	// balancer.Balancer calls as well, and children are called one at a time),
+	// in which calls are guaranteed to come synchronously. Thus, no extra
+	// synchronization is required in this balancer.
+	balancer.ClientConn
+	bOpts balancer.BuildOptions
+
+	// hardcoded child config, graceful switch that wraps a pick_first with no
+	// shuffling enabled.
+	balancerAggregator balancer.Balancer
+
+	cfg *customRRConfig
+
+	// InhibitPickerUpdates determines whether picker updates from the child
+	// forward to cc or not.
+	inhibitPickerUpdates bool
+}
+
+func (crr *customRoundRobin) UpdateClientConnState(state balancer.ClientConnState) error {
+	crrCfg, ok := state.BalancerConfig.(*customRRConfig)
+	if !ok {
+		return balancer.ErrBadResolverState
+	}
+	crr.cfg = crrCfg
+	return crr.balancerAggregator.UpdateClientConnState(balancer.ClientConnState{
+		BalancerConfig: crrCfg.ChildPolicy,
+		ResolverState:  state.ResolverState,
+	})
+}
+
+func (crr *customRoundRobin) ResolverError(err error) {
+	crr.balancerAggregator.ResolverError(err)
+}
+
+// This function is deprecated. SubConn state updates now come through listener
+// callbacks. This balancer does not deal with SubConns directly and has no need
+// to intercept listener callbacks.
+func (crr *customRoundRobin) UpdateSubConnState(sc balancer.SubConn, state balancer.SubConnState) {
+	logger.Errorf("custom_round_robin: UpdateSubConnState(%v, %+v) called unexpectedly", sc, state)
+}
+
+func (crr *customRoundRobin) Close() {
+	crr.balancerAggregator.Close()
+}
+
+// regeneratePicker generates a picker if both child balancers are READY and
+// forwards it upward.
+func (crr *customRoundRobin) UpdateChildState(childStates []balanceraggregator.ChildState) {
+	var readyPickers []balancer.Picker
+	for _, childState := range childStates {
+		if childState.State.ConnectivityState == connectivity.Ready {
+			readyPickers = append(readyPickers, childState.State.Picker)
+		}
+	}
+
+	// For determinism, this balancer only updates it's picker when both
+	// backends of the example are ready. Thus, no need to keep track of
+	// aggregated state and can simply specify this balancer is READY once it
+	// has two ready children. Other balancers can keep track of aggregated
+	// state and interact with errors as part of picker.
+	if len(readyPickers) != 2 {
+		return
```

Comment:
This needs to send a Picker.

I think we need the aggregated state here, too.

Reply:
Agreed for most petiole policies we want to incorporate errors/reflect most recent state to clientconn, but don't you want the picker in this example to be deterministic (i.e. two ready SubConn's which give you the pattern in the README?)